tsconfig.json

```diff
@@ -1,6 +1,7 @@
 {
 	"compilerOptions": {
 		"module": "CommonJS",
+		"esModuleInterop": true,
```

Comment:
Alright, elephant in the room.

ESM advantages (over CommonJS):

Circulars are a lot harder, exports are defined statically without running the code.
Top-level await (TLA).
Async loading, which makes modules be concurrently loaded (faster start-up times).
Much higher safety (you can't import values that don't exist).
Works everywhere, not just in Node.js.
Supports querystrings, which you can read from import.meta.url, this allows you to do a single file that behaves one way or another based on it.
 It's in the language's standard (ECMA 262, also known as ECMAScript) way of defining modules, hence the name, "ECMAScript Modules".
TypeScript uses ESM syntax, if you work with ESM, you have less to learn.
Because it's built into the language's engine, there's no extra overhead nor translation layer to make JS code work, in contrast to CommonJS, which has a lot of code.
ESM is always strict, traditionally you would always need to write 'use strict'; on top to get strict mode's advantages. This also boosts your app's performance.
Can import modules from URLs, and even do so dynamically.
Can export data from sources using asynchronous ways (e.g. dynamic import based on querystring or env parameters, preloading data from a database...) without extra boilerplate.

CJS advantages:

Can import without .js in the domain (unless you use a Node.js flag).
Synchronous loading, comes with a lot of caveats (see above).

If not done in this PR, I would want this update blocked until we move to ESM. It make it SO much nicer to work with.